src/server/pps/server/pipeline_controller.go

```diff
@@ -717,14 +717,15 @@ func (pc *pipelineController) getRC(ctx context.Context, pi *pps.PipelineInfo) (
 		if errCount >= maxErrCount {
 			invalidRCState := errors.Is(err, errTooManyRCs) || errors.Is(err, errStaleRC)
 			if invalidRCState {
-				return pc.restartPipeline(ctx, pi, rc, fmt.Sprintf("could not get RC after %d attempts: %v", errCount, err))
+				restart = fmt.Sprintf("could not get RC after %d attempts: %v", errCount, err)
```

Comment:
i feel like this error message was never accurate, maybe `restart` should just be a bool since it will always be the same string anyway